samples/python/llm/utils/ov_model_classes.py

```diff
@@ -0,0 +1,637 @@
+        dynamic_shapes: bool = True,
+        ov_config: Optional[Dict[str, str]] = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        **kwargs,
+    ):
+        NormalizedConfigManager._conf['mpt'] = NormalizedTextConfig.with_args(num_layers='n_layers', num_attention_heads='n_heads')
+        super().__init__(model, config, device, dynamic_shapes, ov_config, model_save_dir, **kwargs)
+
+    def _reshape(
+        self,
+        model: Model,
+        *args,
+        **kwargs,
+    ):
+        shapes = {}
+        for inputs in model.inputs:
+            shapes[inputs] = inputs.get_partial_shape()
+            if shapes[inputs].rank.get_length() in [2, 3]:
+                shapes[inputs][1] = -1
+            else:
+                if '.key' in inputs.get_any_name():
+                    shapes[inputs][3] = -1
+                else:
+                    shapes[inputs][2] = -1
+        model.reshape(shapes)
+        return model
+
+    def forward(
+        self,
+        input_ids: torch.LongTensor,
+        attention_mask: Optional[torch.LongTensor] = None,
+        past_key_values: Optional[Tuple[Tuple[torch.FloatTensor]]] = None,
+        **kwargs,
+    ) -> CausalLMOutputWithPast:
+        self.compile()
+
+        if self.use_cache and past_key_values is not None:
+            input_ids = input_ids[:, -1:]
+
+        inputs = {}
+        if past_key_values is not None:
+            # Flatten the past_key_values
+            past_key_values = tuple(past_key_value for pkv_per_layer in past_key_values for past_key_value in pkv_per_layer)
+            # Add the past_key_values to the decoder inputs
+            inputs = dict(zip(self.key_value_input_names, past_key_values))
+
+        # Create empty past_key_values for decoder_with_past first generation step
+        elif self.use_cache:
+            shape_input_ids = input_ids.shape
+            num_attention_heads = self.normalized_config.num_attention_heads if self.config.model_type == 'bloom' else 1
+            for input_name in self.key_value_input_names:
+                model_inputs = self.model.input(input_name)
+                shape = model_inputs.get_partial_shape()
+                shape[0] = shape_input_ids[0] * num_attention_heads
+                if shape[2].is_dynamic:
+                    shape[2] = 0
+                if shape[1].is_dynamic:
+                    shape[1] = 0
+                if shape.rank.get_length() == 4 and shape[3].is_dynamic:
+                    shape[3] = 0
+                inputs[input_name] = Tensor(model_inputs.get_element_type(), shape.get_shape())
+
+        inputs['input_ids'] = np.array(input_ids)
+
+        # Add the attention_mask inputs when needed
+        if 'attention_mask' in self.input_names and attention_mask is not None:
+            inputs['attention_mask'] = np.array(attention_mask)
+
+        # Run inference
+        self.request.start_async(inputs, shared_memory=True)
```

Comment:
@eaidova Can you answer the question in https://github.com/openvinotoolkit/openvino/pull/20111/files#r1350183118 from @jiwaszki ?
"Is there any explicit reason to use asynchronous API, any perf benefits?"